src/components/graph-view.js

```diff
@@ -85,24 +86,17 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
   };
 
   static getDerivedStateFromProps(
-    nextProps: IGraphViewProps,
-    prevState: IGraphViewState
+    props: IGraphViewProps,
+    state: IGraphViewState
   ) {
-    const { edges, nodeKey } = nextProps;
-    let nodes = nextProps.nodes;
+    const { edges, nodeKey } = props;
+    const nodes = props.nodes;
     const nodesMap = GraphUtils.getNodesMap(nodes, nodeKey);
     const edgesMap = GraphUtils.getEdgesMap(edges);
 
     GraphUtils.linkNodesAndEdges(nodesMap, edges);
 
-    // Handle layoutEngine on initial render
-    if (prevState.nodes.length === 0 && nextProps.layoutEngine) {
-      const newNodes = nextProps.layoutEngine.adjustNodes(nodes, nodesMap);
```

Comment:
The layout will only occur at `componentDidUpdate`, which doesn't make a difference to the user and allows us to take advantage of async layout engines